src/indexer.ts

```diff
@@ -78,8 +78,37 @@ export class Indexer {
         const error = proxy.unwrapErr();
         if (error.startsWith('[-32000] Server error: DB Not Found Error')) {
           if (blockID < currentBlockHeight) {
-            this.logger.error(error);
-            return; // a skip block, or an unavailable block on a nonarchival node
+            // Handling empty blocks
+            const emptyBlock: EmptyBlock = generateEmptyBlock(
+              blockID as number,
+              contractID.toString(),
+              this.network.chainID,
+              timestamp + 1 // arbitrary timestamp = previous block.timestamp + 1
+            );
+            //if (this.config.debug) console.debug(emptyBlock); // DEBUG
+            const query = sql.insert('block', {
+              chain: this.network.chainID,
+              id: emptyBlock.id,
+              hash: emptyBlock.hash,
+              near_hash: null,
+              timestamp: emptyBlock.timestamp,
+              size: emptyBlock.size,
+              gas_limit: 0, // FIXME: block.gasLimit,
+              gas_used: 0, // FIXME: block.gasUsed,
```

Comment:
There is no reason to retain `FIXME` comments here. Just use `emptyBlock.gasLimit` and `emptyBlock.gasUsed`.